### ISSUE TITLE
Tests fail when not connected to the network

### ISSUE THREAD

**Issue by anthonyryan1** [libgit2/libgit2#3023]
When running the tests without being connected to a network (except for the localhost/loopback interface) the `network::remote` tests fail. Connecting to the network again makes the tests pass.

```
  1) Failure:
network::remote::remotes::single_branch [/var/tmp/portage/dev-libs/libgit2-0.22.2/work/libgit2-0.22.2/tests/network/remote/remotes.c:562]
  Function call failed: (git_clone(&repo, "git://github.com/libgit2/TestGitRepository", "./single-branch", &opts))
  error -1 - Failed to resolve address for github.com: Temporary failure in name resolution

  2) Failure:
network::remote::remotes::restricted_refspecs [/var/tmp/portage/dev-libs/libgit2-0.22.2/work/libgit2-0.22.2/tests/network/remote/remotes.c:582]
  git_clone(&repo, "git://github.com/libgit2/TestGitRepository", "./restrict-refspec", &opts) != GIT_EINVALIDSPEC
  -1 != -12
```

**Comment by ethomson** [comment]
Yeah, these should probably be `online` tests, not `network` tests.  Thanks!

**Comment by cthomas** [comment]
@ethomson @carlosmn @anthonyryan1 
Please see the referenced pull request I submitted for a proposed fix for this issue.

Thanks!

**Comment by cthomas** [comment]
I also just realized that I sent over the pull request from the master branch of my fork instead of a named branch.  Apologies for that, and please let me know if you'd prefer I resubmit the request from a named branch.

**Comment by carlosmn** [comment]
Fixed in #3065
